server/etcdserver/bootstrap_test.go

```diff
@@ -138,3 +150,165 @@ func mockMembersJSON(m []etcdserverpb.Member) string {
 	members, _ := json.Marshal(m)
 	return string(members)
 }
+
+func TestBootstrapBackend(t *testing.T) {
+	tests := []struct {
+		name                  string
+		prepareData           func(config.ServerConfig) error
+		expectedConsistentIdx uint64
+		hasError              bool
+		expectedError         error
+	}{
+		{
+			name:                  "bootstrap backend success: no data files",
+			prepareData:           nil,
+			expectedConsistentIdx: 0,
+			hasError:              false,
+			expectedError:         nil,
+		},
+		{
+			name:                  "bootstrap backend success: have data files and snapshot db file",
+			prepareData:           prepareData,
+			expectedConsistentIdx: 5,
+			hasError:              false,
+			expectedError:         nil,
+		},
+		// TODO(ahrtr): add more test cases
```

Comment:
Can you instead create an issue it should allow anyone to continue the work.

Reply:
Make sense. Just raised a new Jira https://github.com/etcd-io/etcd/issues/13507. 
I will work on it later if nobody work on it.